plugins/modules/inventory_intent.py

```diff
@@ -399,24 +486,36 @@ def mandatory_parameter(self, config):
             It will check the mandatory parameters for adding the devices in Cisco DNA Center.
         """
 
-        device_type = config.get("type", "NETWORK_DEVICE")
+        device_type = self.config[0].get("type", "NETWORK_DEVICE")
 
         mandatory_params_absent = []
         if device_type == "NETWORK_DEVICE":
```

Comment:
can we optimize all the if and elif statements using dictionary?

```
    params_dict = {
        "NETWORK_DEVICE": ["enablePassword", "ipAddress", "password", "snmpUserName", "snmpAuthPassphrase", "snmpPrivPassphrase", "userName"],
        "COMPUTE_DEVICE": ["ipAddress", "httpUserName", "httpPassword", "httpPort", "snmpUserName", "snmpAuthPassphrase", "snmpPrivPassphrase"],
        "MERAKI_DASHBOARD": ["httpPassword"],
        "FIREPOWER_MANAGEMENT_SYSTEM": ["ipAddress", "httpUserName", "httpPassword"],
        "THIRD_PARTY_DEVICE": ["ipAddress", "snmpUserName", "snmpAuthPassphrase", "snmpPrivPassphrase"]
    }

    params_list = params_dict.get(device_type, [])
```

Reply:
yes, I'll address this comment in next PR for resync of devices.